server/activate.go

```diff
@@ -27,78 +30,113 @@ func (p *Plugin) OnActivate() error {
 		return appErr
 	}
 
-	if os.Getenv("CALLS_IS_HANDLER") != "" {
+	cfg := p.getConfiguration()
+	if err := cfg.IsValid(); err != nil {
+		p.LogError(err.Error())
+		return err
+	}
+
+	if cfg.RTCDServiceURL != "" {
+		u, err := url.Parse(cfg.RTCDServiceURL)
+		if err != nil {
+			p.LogError(err.Error())
+			return err
+		}
+
+		clientID := u.User.Username()
+		authKey, _ := u.User.Password()
+		u.User = nil
+
+		client, err := rtcd.NewClient(rtcd.ClientConfig{
+			URL:      u.String(),
+			ClientID: clientID,
+			AuthKey:  authKey,
+		})
+		if err != nil {
+			err = fmt.Errorf("failed to create rtcd client: %w", err)
+			p.LogError(err.Error())
+			return err
+		}
+		if err := client.Connect(); err != nil {
+			err = fmt.Errorf("failed to connect rtcd client: %w", err)
+			p.LogError(err.Error())
+			return err
+		}
+		p.rtcdClient = client
 		go func() {
-			p.LogInfo("calls handler, setting state", "clusterID", status.ClusterId)
-			if err := p.setHandlerID(status.ClusterId); err != nil {
+			for err := range p.rtcdClient.ErrorCh() {
 				p.LogError(err.Error())
-				return
 			}
-			ticker := time.NewTicker(handlerKeyCheckInterval)
-			defer ticker.Stop()
-			for {
-				select {
-				case <-ticker.C:
-					if err := p.setHandlerID(status.ClusterId); err != nil {
-						p.LogError(err.Error())
+		}()
+	} else {
+		if os.Getenv("CALLS_IS_HANDLER") != "" {
```

Comment:
I wonder if there's a way to simplify this nested if-else-if (and the rest of the function below)? I only ask because it's tough for me to tell what's being executed when only looking at the github review interface, which is usually a sign that things have gotten too nested.

Reply:
Hmm.. You could do the go solution of having a `Server` interface and then an `rtcdServer` and a `callsServer` implementation? Although that might cause too much duplication... 0/5, your call of course

Reply:
This needs some good thought cause interfaces would be an added layer of indirection which may just spread things further without actually aiding in understanding what's really going on. At some point we still need to make the distinction between what mode the plugin is operating on and currently we support at least three:


- Hosting calls on all plugin instances.
  - Single handler mode: only a single instance (cluster node) will host all the calls for the whole installation (this is how community operates today).
- Off-loading all calls to external `rtcd` service.

I am thinking of deferring any refactoring to a later point when the whole interaction with `rtcd` has been finalized.